src/index.js

```diff
@@ -25,10 +25,17 @@ async function run () {
   let config = store.get('ipfsConfig')
 
   try {
+    const ipfsd = await setupConnection()
+
+    app.on('quit', () => {
+      ipfsd.stop(err => {
```

Comment:
This is not sync so we are never sure this code will actually be executed. What if the program exits before the callback is called?

Reply:
Another interesting information is:

> Note: On Windows, this event will not be emitted if the app is closed due to a shutdown/restart of the system or a user logout.

So we will basically show the fsck error every time the users logs in. I'll check tomorrow if there are any events on Windows for shutdown/restart/logout.

Reply:
Good point! The reason for the change was to pull the shutdown logic out of the menubar code and up next to the ipfsd creation so you can see where it is created and stopped in the same place. I'm assuming that there are OS specific ways to quit the app that don't involved hitting the power button on the menubar.

Reply:
If I open the webui window and hit `⌘ q` to quit the app on osx, it shuts down the app. So we need to try and stop the ipfsd in in a global quit handler if possible.